### PR TITLE
json-c: remove workaround for BUILD_SHARED_LIBS

### PR DESCRIPTION
**Pull request by Nekto89**
Specify library name and version:  **json-c/all**

Cosmetic changes to make recipe closer to cmake_package template.

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/json-c/all/conanfile.py [hunk 1]
```diff
@@ -70,3 +68,5 @@ def package_info(self):
         self.cpp_info.set_property("cmake_target_name", "json-c::json-c")
         self.cpp_info.set_property("pkg_config_name", "json-c")
         self.cpp_info.libs = collect_libs(self)
+        if self.settings.os in ["Linux", "FreeBSD"]:
+            self.cpp_info.system_libs.append("m")
```
Comment (SpaceIm) [inline recipes/json-c/all/conanfile.py:72]: It doesn't depend on libm AFAIK

File: recipes/json-c/all/test_package/conanfile.py [hunk 2]
```diff
@@ -1,20 +1,20 @@
...
 
-    def layout(self):
-        cmake_layout(self)
-
     def requirements(self):
         self.requires(self.tested_reference_str)
 
+    def layout(self):
+        cmake_layout(self)
+
```
Comment (SpaceIm) [inline recipes/json-c/all/test_package/conanfile.py:17]: Could you keep methods by order execution please? Templates are not Tables of the Law.

File: recipes/json-c/all/conanfile.py (hunk 1 above)
Comment (Nekto89) [reply recipes/json-c/all/conanfile.py:72]: I found it here:
https://github.com/json-c/json-c/blob/d1716fe431694bfc08ad94d2e8d2a32d79b47fa3/CMakeLists.txt#LL98-L100C8

Comment (Nekto89) [reply recipes/json-c/all/conanfile.py:72]: https://github.com/json-c/json-c/pull/182

File: recipes/json-c/all/test_package/conanfile.py (hunk 2 above)
Comment (Nekto89) [reply recipes/json-c/all/test_package/conanfile.py]: done

**Review by StellaSmith — DISMISSED** [review]

File: recipes/json-c/all/test_v1_package/conanfile.py [hunk 3]
```diff
@@ -1,8 +1,9 @@
-from conans import ConanFile, CMake, tools
+from conans import ConanFile, CMake
+from conan.tools.build import cross_building
```
Comment (SpaceIm) [inline recipes/json-c/all/test_v1_package/conanfile.py:2]: I think it was fine. Pure conan v1 usage, without any conan v2 stuff.

Comment (Nekto89) [reply recipes/json-c/all/test_v1_package/conanfile.py]: someone should update template, it's confusing that following it is considered a bad change

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**